core/trino-grammar/src/main/antlr4/io/trino/grammar/sql/SqlBase.g4

```diff
@@ -582,66 +582,70 @@ valueExpression
-    | value=primaryExpression '[' index=valueExpression ']'                               #subscript
-    | identifier                                                                          #columnReference
-    | base=primaryExpression '.' fieldName=identifier                                     #dereference
-    | name=CURRENT_DATE                                                                   #currentDate
-    | name=CURRENT_TIME ('(' precision=INTEGER_VALUE ')')?                                #currentTime
-    | name=CURRENT_TIMESTAMP ('(' precision=INTEGER_VALUE ')')?                           #currentTimestamp
-    | name=LOCALTIME ('(' precision=INTEGER_VALUE ')')?                                   #localTime
-    | name=LOCALTIMESTAMP ('(' precision=INTEGER_VALUE ')')?                              #localTimestamp
-    | name=CURRENT_USER                                                                   #currentUser
-    | name=CURRENT_CATALOG                                                                #currentCatalog
-    | name=CURRENT_SCHEMA                                                                 #currentSchema
-    | name=CURRENT_PATH                                                                   #currentPath
-    | TRIM '(' (trimsSpecification? trimChar=valueExpression? FROM)?
-        trimSource=valueExpression ')'                                                    #trim
-    | TRIM '(' trimSource=valueExpression ',' trimChar=valueExpression ')'                #trim
-    | SUBSTRING '(' valueExpression FROM valueExpression (FOR valueExpression)? ')'       #substring
-    | NORMALIZE '(' valueExpression (',' normalForm)? ')'                                 #normalize
-    | EXTRACT '(' identifier FROM valueExpression ')'                                     #extract
-    | '(' expression ')'                                                                  #parenthesizedExpression
-    | GROUPING '(' (qualifiedName (',' qualifiedName)*)? ')'                              #groupingOperation
-    | JSON_EXISTS '(' jsonPathInvocation (jsonExistsErrorBehavior ON ERROR)? ')'          #jsonExists
+    | EXISTS '(' query ')'                                                                  #exists
+    | CASE operand=expression whenClause+ (ELSE elseExpression=expression)? END             #simpleCase
+    | CASE whenClause+ (ELSE elseExpression=expression)? END                                #searchedCase
+    | CAST '(' expression AS type ')'                                                       #cast
+    | TRY_CAST '(' expression AS type ')'                                                   #cast
+    // the target is a primaryExpression to support PostgreSQL-style casts
+    // of the form <complex expression>::<type>, which are syntactically ambiguous with
+    // static method calls
+    | primaryExpression DOUBLE_COLON identifier ('(' (expression (',' expression)*)? ')')?  #staticMethodCall
```

Comment:
use `type` on the rhs?

AFAIR engines allow  `a::timestamp with time zone` and likely `a::double precision` too

Reply:
to me, "syntactically ambiguous with static method calls" means that `foo::bar(1)` is ambiguous in a sense that it can be two different things

- PostgreSQL-style cast of expression `foo` to type `bar(1)`
- a static method `bar` invocation on type `foo` with parameters `1`

it is surprising, but it seems to me that `foo::bar` is also ambiguous because argument braces are optional

<img width="658" height="291" alt="image" src="https://github.com/user-attachments/assets/9b3d5371-54cb-499c-aa29-6cee71c772df" />

now,
`foo::timestamp` already departs from the spec
`foo::timestamp with time zone` also departs from the spec, but is consistent with `foo::timestamp`. And is not ambiguous.